e2e/wdio/core/tests/calendar.e2e-spec.ts

```diff
@@ -0,0 +1,412 @@
+            checkChangeMonthByNavArrows(programmaticCalendar);
+            checkChangeDateByCalendarOverview(programmaticCalendar, monthBtn);
+            checkChangeDateByCalendarOverview(programmaticCalendar, yearBtn);
+            checkChangeYearByYearRange(programmaticCalendar);
+        });
+
+        it('should check ability to change day with button', () => {
+            const startDay = getText(programmaticCalendar + selectedDays);
+            click(programmaticCalendar + button, 4);
+
+            expect(getText(programmaticCalendar + selectedDays)).not.toEqual(startDay);
+        });
+    });
+
+    describe('monday start calendar example', function() {
+        it('should check calendar starts on Monday', () => {
+            expect(getText(mondayCalendar + calendarDays)).toEqual(mondayStartDate);
+        });
+
+        it('should check calendar selections', () => {
+            checkSingleSelection(mondayCalendar, calendarItem);
+            checkChangeMonthByNavArrows(mondayCalendar);
+            checkChangeDateByCalendarOverview(mondayCalendar, monthBtn);
+            checkChangeDateByCalendarOverview(mondayCalendar, yearBtn, 3);
+            checkChangeYearByYearRange(mondayCalendar);
+        });
+    });
+
+    describe('internationalization example', function() {
+        it('should check calendar selection', () => {
+            checkSingleSelection(internationalCalendar, calendarItem);
+            checkChangeMonthByNavArrows(internationalCalendar);
+            checkChangeDateByCalendarOverview(internationalCalendar, monthBtn);
+            checkChangeDateByCalendarOverview(internationalCalendar, yearBtn, 3);
+            checkChangeYearByYearRange(internationalCalendar);
+        });
+    });
+
+    describe('calendar in reactive form example', function() {
+        it('should check single reactive calendar selections', () => {
+            checkSingleSelection(singleReactiveCalendar, calendarItem);
+        });
+
+        it('should check range reactive calendar selections', () => {
+            checkRangeSelection(rangeReactiveCalendar);
+        });
+
+        it('should check invalid date buttons', () => {
+            expect(isElementClickable(reactiveCalendarExamples + button, 4)).toBe(true);
+            expect(isElementClickable(reactiveCalendarExamples + button, 9)).toBe(true);
+        });
+    });
+
+    describe('check orientation', function() {
+        it('should check orientation', () => {
+            calendarPage.checkRtlSwitch();
+        });
+    });
+
+    function checkChangeMonthByNavArrows(calendar: string): void {
+        const startMonth = getText(calendar + monthBtn);
+
+        scrollIntoView(calendar + monthBtn);
+        click(calendar + leftArrowBtn);
+
+        expect(getText(calendar + monthBtn)).not.toEqual(startMonth);
+        click(calendar + rightArrowBtn);
+
+        expect(getText(calendar + monthBtn)).toEqual(startMonth);
+        click(calendar + rightArrowBtn);
```

Comment:
if rightArrowBtn isn't work it will fail on previous expect, third check is not needed.
If you are checking that user can actually navigate through past months, then you can create separate check and make this function (checkChangeMonthByNavArrows) with one less reason to fail.